diff -urN ../guile-1.3.4/ChangeLog ./ChangeLog
--- ../guile-1.3.4/ChangeLog	Sat Sep 25 19:26:50 1999
+++ ./ChangeLog	Mon Mar 27 15:46:54 2000
@@ -1,3 +1,12 @@
+2000-03-24    <janneke@gnu.org>
+ 
+	* guile-config/cross-guile-config.in: Simple shell script to
+ 	take over guile-config's functionality for cross-compilation.
+ 	It will be installed as ${build}-guile-config.
+ 
+ 	* guile-config/Makefile.am: Provide for cross-guile-config.in
+ 	and its targets.
+ 
 1999-09-25  Jim Blandy  <jimb@savonarola.red-bean.com>
 
 	* Guile 1.3.4 released.
diff -urN ../guile-1.3.4/guile-config/Makefile.am ./guile-config/Makefile.am
--- ../guile-1.3.4/guile-config/Makefile.am	Mon Oct 19 17:24:31 1998
+++ ./guile-config/Makefile.am	Mon Mar 27 15:47:43 2000
@@ -20,9 +20,9 @@
 ##   to the Free Software Foundation, Inc., 59 Temple Place, Suite
 ##   330, Boston, MA 02111-1307 USA
 
-bin_SCRIPTS=guile-config
-CLEANFILES=guile-config
-EXTRA_DIST=guile-config.in
+bin_SCRIPTS=guile-config @build@-guile-config
+CLEANFILES=guile-config @build@-guile-config
+EXTRA_DIST=guile-config.in cross-guile-config.in
 
 ## We use @-...-@ as the substitution brackets here, instead of the
 ## usual @...@, so autoconf doesn't go and substitute the values
@@ -34,6 +34,15 @@
 	    -e s:@-GUILE_VERSION-@:${GUILE_VERSION}:
 	chmod +x guile-config.tmp
 	mv guile-config.tmp guile-config
+
+@build@-guile-config: cross-guile-config.in
+	rm -f $<.tmp
+	sed < ${srcdir}/$< > $<.tmp \
+	    -e s:@-SHELL-@:${SHELL}: \
+	    -e s:@-GUILE_VERSION-@:${GUILE_VERSION}:\
+	    -e s:@-PREFIX-@:${prefix}:
+	chmod +x $<.tmp
+	mv $<.tmp $@
 
 ## Get rid of any copies of the configuration script under the old
 ## name, so people don't end up running ancient copies of it.
diff -urN ../guile-1.3.4/guile-config/cross-guile-config.in ./guile-config/cross-guile-config.in
--- ../guile-1.3.4/guile-config/cross-guile-config.in	Thu Jan  1 01:00:00 1970
+++ ./guile-config/cross-guile-config.in	Mon Mar 27 15:46:36 2000
@@ -0,0 +1,16 @@
+#!@-SHELL-@
+# build-guile-config.in
+case $1 in
+	--version)
+		echo "@-GUILE_VERSION-@"
+		exit 0
+		;;
+	compile)
+		echo "-I @-PREFIX-@/include/guile"
+		exit 0
+		;;
+	link)
+		echo "-L @-PREFIX-@/lib -lguile"
+		exit 0
+		;;
+esac
diff -urN ../guile-1.3.4/libguile/dynl-dl.c ./libguile/dynl-dl.c
--- ../guile-1.3.4/libguile/dynl-dl.c	Sun Aug 29 20:03:26 1999
+++ ./libguile/dynl-dl.c	Mon Mar 27 15:57:24 2000
@@ -45,7 +45,9 @@
    Author: Aubrey Jaffer
    Modified for libguile by Marius Vollmer */
 
+#ifndef __CYGWIN__
 #include <dlfcn.h>
+#endif
 
 #ifdef RTLD_LAZY	/* Solaris 2. */
 #  define DLOPEN_MODE	RTLD_LAZY
diff -urN ../guile-1.3.4/libguile/filesys.c ./libguile/filesys.c
--- ../guile-1.3.4/libguile/filesys.c	Sun Sep 12 05:47:39 1999
+++ ./libguile/filesys.c	Mon Mar 27 15:57:24 2000
@@ -138,7 +138,9 @@
 	fdes = SCM_INUM (object);
       else
 	fdes = SCM_FPORT_FDES (object);
+#ifndef __CYGWIN__
       SCM_SYSCALL (rv = fchown (fdes, SCM_INUM (owner), SCM_INUM (group)));
+#endif
     }
   else
     {
