diff -urN ../guile-core/libguile/Makefile.am ./libguile/Makefile.am
--- ../guile-core/libguile/Makefile.am	Sat Oct 10 13:00:59 1998
+++ ./libguile/Makefile.am	Wed Oct 21 10:58:56 1998
@@ -10,7 +10,7 @@
 ETAGS_ARGS = --regex='/SCM_\(GLOBAL_\)?\(PROC\|PROC1\|SYMBOL\|VCELL\|CONST_LONG\).*\"\([^\"]\)*\"/\3/'
 
 lib_LTLIBRARIES = libguile.la
-bin_PROGRAMS = guile
+bin_PROGRAMS = guile.exe
 
 guile_SOURCES = guile.c
 guile_LDADD = libguile.la ${THREAD_LIBS_LOCAL}
diff -urN ../guile-core/libguile/Makefile.in ./libguile/Makefile.in
--- ../guile-core/libguile/Makefile.in	Sat Oct 10 13:00:59 1998
+++ ./libguile/Makefile.in	Wed Oct 21 10:59:02 1998
@@ -94,7 +94,7 @@
 ETAGS_ARGS = --regex='/SCM_\(GLOBAL_\)?\(PROC\|PROC1\|SYMBOL\|VCELL\|CONST_LONG\).*\"\([^\"]\)*\"/\3/'
 
 lib_LTLIBRARIES = libguile.la
-bin_PROGRAMS = guile
+bin_PROGRAMS = guile.exe
 
 guile_SOURCES = guile.c
 guile_LDADD = libguile.la ${THREAD_LIBS_LOCAL}
diff -urN ../guile-core/libguile/dynl-dl.c ./libguile/dynl-dl.c
--- ../guile-core/libguile/dynl-dl.c	Mon Mar 30 23:03:30 1998
+++ ./libguile/dynl-dl.c	Wed Oct 21 09:41:55 1998
@@ -45,8 +45,6 @@
    Author: Aubrey Jaffer
    Modified for libguile by Marius Vollmer */
 
-#include <dlfcn.h>
-
 #ifdef RTLD_LAZY	/* Solaris 2. */
 #  define DLOPEN_MODE	RTLD_LAZY
 #else
diff -urN ../guile-core/libguile/filesys.c ./libguile/filesys.c
--- ../guile-core/libguile/filesys.c	Fri Oct 09 20:51:15 1998
+++ ./libguile/filesys.c	Wed Oct 21 10:18:12 1998
@@ -138,15 +138,19 @@
 	  if (fdes == -1)
 	    scm_syserror (s_chown);
 	}
+#if 0
       SCM_SYSCALL (rv = fchown (fdes, SCM_INUM (owner), SCM_INUM (group)));
+#endif
     }
   else
     {
       SCM_ASSERT (SCM_NIMP (object) && SCM_ROSTRINGP (object),
 		  object, SCM_ARG1, s_chown);
       SCM_COERCE_SUBSTR (object);
+#if 0
       SCM_SYSCALL (rv = chown (SCM_ROCHARS (object),
 			       SCM_INUM (owner), SCM_INUM (group)));
+#endif
     }
   if (rv == -1)
     scm_syserror (s_chown);
@@ -557,14 +561,18 @@
 	  if (fdes == -1)
 	    scm_syserror (s_truncate_file);
 	}
+#if 0
       SCM_SYSCALL (rv = ftruncate (fdes, csize));
+#endif
     }
   else
     {
       SCM_ASSERT (SCM_NIMP (object) && SCM_ROSTRINGP (object),
 		  object, SCM_ARG1, s_chown);
       SCM_COERCE_SUBSTR (object);
+#if 0
       SCM_SYSCALL (rv = truncate (SCM_ROCHARS (object), csize));
+#endif
     }
   if (rv == -1)
     scm_syserror (s_truncate_file);
